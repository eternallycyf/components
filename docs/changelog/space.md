---
title: Space
nav:
  title: changelog
  order: 999
tocDepth: 2
---

## Space

<embed src="../../packages/space/CHANGELOG.md"></embed>
