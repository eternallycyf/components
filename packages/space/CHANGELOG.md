# Changelog

## [Version&nbsp;1.1.0](https://github.com/eternallycyf/components/compare/@ims-view/space@1.0.1...@ims-view/space@1.1.0)

<sup>Released on **2024-04-30**</sup>

#### ✨ 新特性

- Add Message.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- Add Message ([936f2c6](https://github.com/eternallycyf/components/commit/936f2c6))

</details>

### [Version&nbsp;1.0.1](https://github.com/eternallycyf/components/compare/@ims-view/space@1.0.0...@ims-view/space@1.0.1)

<sup>Released on **2024-04-28**</sup>

#### 🐛 修复

- Keep react version.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's fixed

- Keep react version ([fee7e8d](https://github.com/eternallycyf/components/commit/fee7e8d))

</details>

## Version&nbsp;1.0.0

<sup>Released on **2024-02-10**</sup>

#### ✨ 新特性

- **@ims-view/space**: Add Space.

<br/>

<details>
<summary><kbd>Improvements and Fixes</kbd></summary>

#### What's improved

- **@ims-view/space**: Add Space ([b1e539d](https://github.com/eternallycyf/components/commit/b1e539d))

</details>
