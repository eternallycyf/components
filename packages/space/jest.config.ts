import { Config } from '@umijs/test';
import base from '../../jest.config.base';

const packageName = '@ims-view/space';

const root = '<rootDir>/packages/space';

const config: Config.InitialOptions = {
  ...base,
  rootDir: '../..',
  roots: [root],
  displayName: packageName,
};

export default config;
