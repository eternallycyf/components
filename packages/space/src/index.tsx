export { default as Space } from './Space';
export { ConfigProvider } from './Space/ConfigProvider';
export * from './Space/interface';
