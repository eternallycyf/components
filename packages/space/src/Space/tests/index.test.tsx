import { ConfigProvider, Space } from '@ims-view/space';
import { render } from '@testing-library/react';
import React from 'react';

describe('Space', () => {
  it('should render width empty children', () => {
    const { container } = render(<Space />);

    expect(container.children.length).toBe(1);
  });

  it('should render width ConfigProvider', () => {
    const { container } = render(
      <ConfigProvider space={{ size: 'large' }}>
        <Space>
          <span>1</span>
          <span>2</span>
        </Space>
        <Space size="middle">
          <span>1</span>
          <span>2</span>
        </Space>
        <Space size="large">
          <span>1</span>
          <span>2</span>
        </Space>
      </ConfigProvider>,
    );

    expect(container.children).toMatchSnapshot();
  });
});
