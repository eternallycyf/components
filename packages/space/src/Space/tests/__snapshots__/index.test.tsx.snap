// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should render width ConfigProvider 1`] = `
HTMLCollection [
  <div
    class="space space-horizontal space-align-center"
    style="column-gap: 24px; row-gap: 24px;"
  >
    <div
      class="space-item"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
  <div
    class="space space-horizontal space-align-center"
    style="column-gap: 16px; row-gap: 16px;"
  >
    <div
      class="space-item"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
  <div
    class="space space-horizontal space-align-center"
    style="column-gap: 24px; row-gap: 24px;"
  >
    <div
      class="space-item"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
]
`;
